### PR TITLE
fix(Yoga): Compile Yoga from source

### LINKED ISSUE #1522: Use the same version of Yoga as the corresponding React Native release
For a given release (e.g. 0.45.0), React Native Windows should use the same exact version of Yoga as React Native. This is important so that apps which are targeting multiple platforms (e.g. UWP, iOS, Android) will have identical layouts across platforms.

Adam Comella
Microsoft Corp.

### PR DESCRIPTION
**Pull request by rozele**
In order to stay better aligned with the version of Yoga used by iOS and Android, this changeset introduces a submodule pointing to the Facebook.Yoga repo.

Fixes #1522

### CODE REVIEW

**Comment by rozele** [comment]
TODO: fixup CLI and Yoga dependency for WPF.

**Comment by rozele** [comment]
Another thing to consider - let's move the Windows specific code into this repo so we can just point to the Yoga C++ code shipped in react-native (once our MSBuild fixes are integrated)

File: local-cli/generator-windows/templates/proj/MyApp.sln [hunk 1]
```diff
@@ -142,6 +150,54 @@ EndGlobalSection
 		{7596216B-669C-41F8-86DA-F3637F6545C0}.ReleaseBundle|x64.Build.0 = Release|x64
 		{7596216B-669C-41F8-86DA-F3637F6545C0}.ReleaseBundle|x86.ActiveCfg = Release|x86
 		{7596216B-669C-41F8-86DA-F3637F6545C0}.ReleaseBundle|x86.Build.0 = Release|x86
+		{D6477AD6-9EDD-4EBA-8C60-CD31F77C52C6}.Debug|ARM.ActiveCfg = Debug|ARM
```
Comment (rozele) [inline local-cli/generator-windows/templates/proj/MyApp.sln]: Delete

**Comment by rigdern** [comment]
@rozele Can you elaborate on the plan for Yoga consumption? Some questions I have:
  - When a React Native Windows app is built, does it build Yoga from source?
  - How do we ensure a given release of React Native Windows uses the exact same version of Yoga as the corresponding release of React Native iOS/Android?

**Comment by rozele** [comment]
@rigdern - Yoga is built from source with RNW, both

**Comment by rigdern** [comment]
@rozele Will both the react-native-windows repo and the yoga repo contain the .vcxproj and the C# files? Or are we deleting them from the yoga repo? If we're deleting them, did you already speak with the yoga team about this?

**Comment by rozele** [comment]
@matthargett I think you might be right. I need to add some compile time constants for ARM builds.

**Comment by rozele** [comment]
@rigdern - I don't know if anyone is using the C# in Yoga. But that's a good point, maybe we should continue to rely on the submodule so we can take advantage of community contributions to the C# wrapper. I think some people are using it for Unity.

**Comment by rigdern** [comment]
@rozele Okay. Is the new plan to consume the Yoga C++ code from the react-native repo and the Yoga C# code from the yoga repo?

**Pull request merged**